native_lib/DDGSyncCrypto.c

```diff
@@ -0,0 +1,150 @@
+
+#include <string.h>
+
+#include "DDGSyncCrypto.h"
+#include "sodium.h"
+
+// Seems mad that you still need to define this?!
+#define min(x, y) (x < y) ? x : y
```

Comment:
`x` and `y` should have parenthesis in the preprocessor definition, to prevent incorrect evaluation order when certain operators are used. 

e.g.

`min(3 | 4, 1 | 2);`

Will be evaluated in the following order:

`(3 | (4 < 1) | 2) ? (3 | 4) : (1 : 2)`

Resulting in a return value of `7`, when the correct answer is `3`. See the example here: 
https://godbolt.org/z/cvq34PbEh

https://en.cppreference.com/w/c/language/operator_precedence

Reply:
@MarkIngramUK thanks for the comment, do you fancy opening a PR for this?